### ISSUE TITLE
`CREATE TABLE` fails at first attempt, succeeds at second. No issue with `CREATE TEMPORARY TABLE`

### ISSUE THREAD

**Issue by Patechoc** [ibmdb/python-ibmdbsa#93]
### First attempt creating the table `my_schema.TMP_EMPTY_TABLE_SA` fails

```python
import sqlalchemy as sa

db2_conn = ("db2+ibm_db://username:password@servername:port/database")
engine = sa.create_engine(db2_conn)
cnxn = engine.connect()

try:
    sql = (
        "CREATE TEMPORARY TABLE my_schema.TMP_EMPTY_TABLE_SA ( ColOne int, ColTwo varchar(255), "
        "ColThree varchar(255), ColFour varchar(255), ColFive varchar(255) );"
    )
    response = cnxn.execute(sql)
    print("SUCCESS CREATING TABLE")
except Exception as e:
    print("FAILED TO CREATE A TABLE")
    print(e)
```

```shell
2021-02-09 15:01:58,931 INFO sqlalchemy.engine.base.Engine CREATE TABLE my_schema.TMP_EMPTY_TABLE_SA ( ColOne int, ColTwo varchar(255), ColThree varchar(255), ColFour varchar(255), ColFive varchar(255) );
2021-02-09 15:01:58,931 INFO sqlalchemy.engine.base.Engine ()
2021-02-09 15:01:59,028 INFO sqlalchemy.engine.base.Engine ROLLBACK
FAILED TO CREATE A TABLE
(ibm_db_dbi.OperationalError) ibm_db_dbi::OperationalError: Exception('Statement Execute Failed: [IBM][CLI Driver][DB2/LINUXX8664] SQL0142N  The SQL statement is not supported. SQLCODE=-142')
[SQL: CREATE TABLE my_schema.TMP_EMPTY_TABLE_SA ( ColOne int, ColTwo varchar(255), ColThree varchar(255), ColFour varchar(255), ColFive varchar(255) );]
(Background on this error at: http://sqlalche.me/e/13/e3q8)
```

## Second attempt creating the table `my_schema.TMP_EMPTY_TABLE_SA` succeeds

```shell
2021-02-09 15:08:24,234 INFO sqlalchemy.engine.base.Engine CREATE TABLE my_schema.TMP_EMPTY_TABLE_SA ( ColOne int, ColTwo varchar(255), ColThree varchar(255), ColFour varchar(255), ColFive varchar(255) );
2021-02-09 15:08:24,235 INFO sqlalchemy.engine.base.Engine ()
2021-02-09 15:08:24,316 INFO sqlalchemy.engine.base.Engine COMMIT
SUCCESS CREATING TABLE
```

## `CREATE TEMPORARY TABLE`  succeeds at first attempt

```python 
...
try:
    sql = (
        "CREATE TEMPORARY TABLE my_schema.TMP_EMPTY_TABLE_SA ( ColOne int, ColTwo varchar(255), "
        "ColThree varchar(255), ColFour varchar(255), ColFive varchar(255) );"
    )
...
```

```shell
2021-02-09 15:14:58,655 INFO sqlalchemy.engine.base.Engine CREATE TEMPORARY TABLE my_schema.TMP_EMPTY_TABLE_SA ( ColOne int, ColTwo varchar(255), ColThree varchar(255), ColFour varchar(255), ColFive varchar(255) );
2021-02-09 15:14:58,655 INFO sqlalchemy.engine.base.Engine ()
2021-02-09 15:14:58,678 INFO sqlalchemy.engine.base.Engine COMMIT
SUCCESS CREATING TABLE
```

**Comment by Patechoc** [comment]
I get the exact same issue with `CREATE VIEW...`: first attempt fails, but running my script a second time creates the VIEW!?

Would you know any specific configuration which could be responsible for such behavior?

**Comment by amukherjee28** [comment]
Hi,

I am working in reproducing the same.

Could you please help me with the python version used while running these applications.

Also the environment Windows/Linux/Mac.

I guess the server in question is Db2/Luw?

Thanks.

**Comment by Patechoc** [comment]
Thanks a lot!

I am running python 3.8 them through a Docker container on Windows.

```shell
root@0158c600e1bc:/opt/project/great_expectations# python --version
Python 3.8.7
root@0158c600e1bc:/opt/project/great_expectations# uname -a
Linux 0158c600e1bc 4.19.121-linuxkit #1 SMP Tue Dec 1 17:50:32 UTC 2020 x86_64 GNU/Linux
```

I am not sure about the version of DB2. I have access to it through DBeaver too, but I am not admin.
Is there a way to find out the type of server (through SqlAlchemy or directly with ibm_db)?

**Comment by amukherjee28** [comment]
hi @Patechoc 

Can you add these lines in your application code to get the DATABASE DETAILS and send me the same.

dbms_name = getattr(conn.connection,'dbms_name','None')
print("DATABASE : " + dbms_name)
dbms_version = getattr(conn.connection,'dbms_ver','None')
print("DATABASE VERSION : " + dbms_version)

We need to understand your server type.

Because from my end I am not facing the similar issue as yours.

Thanks.

**Comment by Patechoc** [comment]
Hi @amukherjee28 

Here are the info about our Db2 instance:

```
DATABASE : DB2/LINUXX8664
DATABASE VERSION : 11.01.0900
```

Thank you very much for your attempt to reproduce the issue 👍 

I found out today that the problem was purely related to "not enough permissions".

- I sat up a local instance running in Docker and I have **NOT** been able to reproduce the issue I reported above.
- Meaning all these SQL queries went fine running on this local IBM Db2 instance in which I probably have superuser permissions :)
- I therefore contacted the admins of our DB instance who increased my permissions and the initial problem disappeared there too.

I can not give you the details of my former permissions which I don't know, but I have been told I had Admin rights now on our corporate test instance, and this solved everything 🥳 

Thanks a lot again for your quick response (much appreciated) and I apologize for having reported a false alarm.

Have a nice weekend!
